### PR TITLE
evm config change helper and transaction building logic.

### PR DESCRIPTION
**Pull request by bflood-strike**

### CODE REVIEW

File: Censo/EvmPolicyHelper.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        return "0x0"
+    }
+    
+    func applyTransaction(_ tx: SafeTx) -> Policy {
+        var currentOwners = owners.map { $0 }
+        var currentThreshold = threshold
+        switch tx {
+        case .swapOwner(let prev, let old, let new):
+            let ownerIndex = currentOwners.firstIndex(of: old)
+            assert(ownerIndex != nil)
```
Comment (bholzman) [inline Censo/EvmPolicyHelper.swift:56]: I believe these assert()ions are only active in debug builds, not Release builds. I'm thinking if one of these conditions were violated somehow, we'd want even a Release build to error out.

File: Censo/EvmWhitelistHelper.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        currentAddresses = addresses.map( { $0.lowercased() })
+        targetAddresses = targetDestinations.map( { $0.address.lowercased() })
+    }
+    
+    private static let censoGuard = "CensoGuard"
+    private static let censoTransfersOnlyGuard = "CensoTransfersOnlyGuard"
+    private static let censoTransfersOnlyWhitelistingGuard = "CensoTransfersOnlyWhitelistingGuard"
+    private static let censoWhitelistingGuard = "CensoWhitelistingGuard"
+    
+    static func getTargetGuardAddress(currentGuardAddress: String, whitelistEnabled: Bool?, dappsEnabled: Bool?, guardAddresses: [ContractNameAndAddress]) -> String {
```
Comment (bholzman) [inline Censo/EvmWhitelistHelper.swift:53]: I'm thinking that for security we want the actual guard addresses to be baked in to the build ... would have to be by chain, and would be different in different environments. So I guess a bunch of application properties?

**Review by bholzman — APPROVED** [review]
Looks great!

File: Censo/EvmConfigTransactionBuilder.swift [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            value: Bignum(0),
+            data: setGuardTx(guardAddress: guardAddress),
+            operation: .call)
+    }
+    
+    // updating whitelist
+    static func getUpdateWhitelistExecutionFromModuleDataSafeHash(walletAddress: String, addsOrRemoves: [Data], signingData: EthereumTransaction) -> Data {
+        return EvmTransactionUtil.computeSafeTransactionHash(
+            chainId: signingData.chainId,
+            safeAddress: signingData.vaultAddress ?? "0x0",
```
[the remark this answers is not in the document]
Comment (ata-peppered) [reply Censo/EvmConfigTransactionBuilder.swift:62]: is it possible to make the property non optional on `EthereumTransaction`?

**Review by ata-peppered — APPROVED** [review]

File: Censo/EvmPolicyHelper.swift (hunk 1 above)
Comment (bflood-strike) [reply Censo/EvmPolicyHelper.swift]: made them throws

File: Censo/EvmWhitelistHelper.swift (hunk 2 above)
Comment (bflood-strike) [reply Censo/EvmWhitelistHelper.swift:50]: I'll add a ticket - for prerelease every reset these guards are a different address I believe. So for prod we can have have configs built in once we know the guard addresses

File: Censo/EvmConfigTransactionBuilder.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (bflood-strike) [reply Censo/EvmConfigTransactionBuilder.swift]: throwing an error on this for now.  @ata-peppered we can look to make it required in the model - I'll add a ticket to make this change in brooklyn.

**Pull request merged**